processor/metricstransformprocessor/metrics_transform_processor_test.go

```diff
@@ -21,301 +21,171 @@ import (
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer/consumerdata"
 	"go.opentelemetry.io/collector/consumer/pdatautil"
 	etest "go.opentelemetry.io/collector/exporter/exportertest"
+	"go.uber.org/zap"
 )
 
-type metricsTransformTest struct {
-	name       string // test name
-	transforms []Transform
-	inMN       []string // input Metric names
-	outMN      []string // output Metric names
-	inLabels   []string
-	outLabels  []string
-}
+func TestMetricsTransformProcessor(t *testing.T) {
+	for _, test := range standardTests {
+		t.Run(test.name, func(t *testing.T) {
+			// next stores the results of the aggregation metric processor.
+			next := &etest.SinkMetricsExporter{}
 
-var (
-	initialMetricNames = []string{
-		"metric1",
-		"metric5",
-	}
+			mtp := newMetricsTransformProcessor(next, zap.NewExample(), test.transforms)
+			assert.NotNil(t, mtp)
 
-	outMetricNamesUpdateSingle = []string{
-		"metric1/new",
-		"metric5",
-	}
-	outMetricNamesUpdateMultiple = []string{
-		"metric1/new",
-		"metric5/new",
-	}
+			caps := mtp.GetCapabilities()
+			assert.Equal(t, true, caps.MutatesConsumedData)
+			ctx := context.Background()
+			assert.NoError(t, mtp.Start(ctx, nil))
 
-	outMetricNamesInsertSingle = []string{
-		"metric1",
-		"metric5",
-		"metric1/new",
-	}
+			// construct metrics data to feed into the processor
+			md := consumerdata.MetricsData{Metrics: test.in}
 
-	outMetricNamesInsertMultiple = []string{
-		"metric1",
-		"metric5",
-		"metric1/new",
-		"metric5/new",
-	}
+			// process
+			cErr := mtp.ConsumeMetrics(
+				context.Background(),
+				pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
+					md,
+				}),
+			)
+			assert.NoError(t, cErr)
 
-	initialLabels = []string{
-		"label1",
-		"label2",
-	}
+			// get and check results
+			got := next.AllMetrics()
+			require.Equal(t, 1, len(got))
+			gotMD := pdatautil.MetricsToMetricsData(got[0])
+			require.Equal(t, 1, len(gotMD))
+			actualOut := gotMD[0].Metrics
+			require.Equal(t, len(test.out), len(actualOut))
 
-	outLabels = []string{
-		"label1/new",
-		"label2",
-	}
+			for idx, out := range test.out {
+				assert.Equal(t, *out, *actualOut[idx])
+			}
 
-	validUpateLabelOperation = Operation{
-		Action:   UpdateLabel,
-		Label:    "label1",
-		NewLabel: "label1/new",
+			assert.NoError(t, mtp.Shutdown(ctx))
```

Comment:
Doesn't really matter for this test, but would be more idiomatic to do `defer func() { assert.NoError(t, mtp.Shutdown(ctx)) }()` immediately after `assert.NoError(t, mtp.Start(ctx, nil))`